codegen/core/src/main/scala/com/lightbend/akkasls/codegen/SourceGeneratorUtils.scala

```diff
@@ -80,7 +80,8 @@ object SourceGeneratorUtils {
   def typeImport(fullyQualifiedName: FullyQualifiedName): String = {
     val name =
       if (fullyQualifiedName.parent.javaMultipleFiles) fullyQualifiedName.name
-      else fullyQualifiedName.parent.javaOuterClassname
+      else if (fullyQualifiedName.parent.javaOuterClassname.nonEmpty) fullyQualifiedName.parent.javaOuterClassname
+      else fullyQualifiedName.name
```

Comment:
what will happen if `java_outer_classname` is defined when generating Scala? You said that ScalaPB don't have that option so maybe things will be totally wrong if that is accidentally defined?

Ignore and fallback to right defaults for Scala or fail fast if `java_outer_classname` is defined for Scala?

Reply:
This code is shared, so we don't "know" if we're generating Scala here.

We can make sure in the Scala `FullyQualifiedNameExtractor` that we never populate `javaOuterClassname` there, but beyond that I'm not sure there's an easy way to add more safeguards. One thing we could do is remove `PackageNaming.from(Descriptor)` and make sure this is only done in `FullyQualifiedNameExtractor`, but I haven't dared to pull that thread yet ;).

(the ideal solution IMO would be to remove `javaOuterClassname` from `FullyQualifiedName` entirely and interpret it inside the Java `FullyQualifiedNameExtractor` already, but that might not be possible/easy)